crates/eth-rpc/src/api/net_api.rs

```diff
@@ -0,0 +1,22 @@
+use jsonrpsee::core::RpcResult as Result;
+use jsonrpsee::proc_macros::rpc;
+use reth_primitives::U64;
+use reth_rpc_types::PeerCount;
+
+// TODO: Define and implement of methods of Net API
+#[rpc(server, namespace = "net")]
+#[async_trait]
+pub trait NetApi {
+    /// Returns the protocol version encoded as a string.
+    #[method(name = "version")]
+    fn version(&self) -> Result<U64>;
+
+    /// Returns number of peers connected to node.
+    #[method(name = "peerCount")]
+    fn peer_count(&self) -> Result<PeerCount>;
+
+    /// Returns true if client is actively listening for network connections.
+    /// Otherwise false.
+    #[method(name = "listening")]
+    fn is_listening(&self) -> Result<bool>;
```

Comment:
why is this `is_listening` and not `listening`?

Reply:
I took the function name from my investigation into [reth](https://github.com/paradigmxyz/reth/blob/15781beda869c5db682bb408d8228b90045d3471/crates/rpc/rpc-api/src/net.rs#L19), the name made sense to me as its return type is a boolean.

But, I think `listening` makes sense as well, I have refactored.